tests/sha1.cc

```diff
@@ -74,17 +75,97 @@ TEST_P(Sha1, KAT)
                              << bytes_to_hex(digest) << std::endl;
 }
 
+// ----- EverCrypt -------------------------------------------------------------
+
+typedef EverCryptSuite<TestCase> EverCryptSuiteTestCase;
+
+TEST_P(EverCryptSuiteTestCase, HashTest)
+{
+  EverCryptConfig config;
+  TestCase test;
+  tie(config, test) = this->GetParam();
+
+  {
+    bytes got_digest(
+      Hacl_Hash_Definitions_hash_len(Spec_Hash_Definitions_SHA1));
+
+    EverCrypt_Hash_hash(Spec_Hash_Definitions_SHA1,
+                        got_digest.data(),
+                        test.msg.data(),
+                        test.msg.size());
+    ASSERT_EQ(test.md, got_digest);
+  }
+
+  {
+    bytes got_digest(
+      Hacl_Hash_Definitions_hash_len(Spec_Hash_Definitions_SHA1));
+
+    // Use a typedef?
+    Hacl_Streaming_Functor_state_s___EverCrypt_Hash_state_s____* state =
```

Comment:
The generation should do nicer names here ... Feel free to file something for hacl upstream.

Reply:
Noted. Let's discuss it first in our API session.